src/components/ItemList.js

```diff
@@ -1,5 +1,43 @@
+import { collection, query, where, onSnapshot } from 'firebase/firestore';
+import { useEffect, useState } from 'react';
+import { db } from '../lib/firebase.js';
+import { Link } from 'react-router-dom';
+
 const ItemList = () => {
-  return <h1>Item List</h1>;
+  const [docs, setDocs] = useState([]);
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    const q = query(
+      collection(db, 'shopping-list'),
+      where('token', '==', token),
+    );
+    const unsubscribe = onSnapshot(q, (querySnapshot) => {
+      const items = [];
+      querySnapshot.forEach((doc) => {
+        items.push(doc.data());
+      });
+      setDocs(items);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, []);
+
+  return (
+    <>
+      <h2>Smart Shopping List</h2>
+      <ul>
+        {docs.length > 0 ? (
```

Comment:
This is 100% valid. For a fun exercise you could play with this in a javascript/browser console. Since a value of `docs.length` value of `0` will be considered _falsy_, you could omit `> 0` here since the `docs.length` of `0` will evaluate to false and have the same effect in the ternary. Again, this is 100% valid and there's a counter argument to mine that says your code is more specific/clear, taking away any ambiguity to what the ternary is comparing!

Reply:
Always helpful to learn about the different approaches/arguments. Thx!